src/main.cpp

```diff
@@ -2121,64 +2121,60 @@ bool GetAddressUnspent(const uint160& addressHash, int type,
  * Return transaction in txOut, and if it was found inside a block, its hash is placed in hashBlock.
  * If blockIndex is provided, the transaction is fetched from the corresponding block.
  */
-bool GetTransaction(const uint256& hash, CTransaction& txOut, const Consensus::Params& consensusParams, uint256& hashBlock, bool fAllowSlow, CBlockIndex* blockIndex)
+bool GetTransaction(const uint256& hash, CTransaction& txOut, const Consensus::Params& consensusParams, uint256& hashBlock, bool fAllowSlow, const CBlockIndex* blockIndex)
 {
-    CBlockIndex* pindexSlow = blockIndex;
-
     LOCK(cs_main);
 
-    if (!blockIndex) {
-        std::shared_ptr<const CTransaction> ptx = mempool.get(hash);
-        if (ptx)
-        {
-            txOut = *ptx;
-            return true;
-        }
+    std::shared_ptr<const CTransaction> ptx = mempool.get(hash);
+    if (ptx) {
+        txOut = *ptx;
+        return true;
+    }
```

Comment:
Is it possible to do this outside `cs_main`? (Note that `mempool.get` locks internally on the mempool lock.)